src/Symfony/Component/HttpKernel/EventListener/SerializeControllerResultListener.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\Serialize;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Serializer\SerializerInterface;
+
+final class SerializeControllerResultListener implements EventSubscriberInterface
+{
+    private const DEFAULT_FORMAT = 'json';
+
+    public function __construct(private readonly ?SerializerInterface $serializer)
+    {
+    }
+
+    public function onView(ViewEvent $event): void
+    {
+        /** @var Serialize[] $attributes */
+        $attributes = $event->controllerArgumentsEvent->getAttributes()[Serialize::class] ?? [];
+        if (!$attributes) {
+            return;
+        }
+
+        if (!$this->serializer) {
+            throw new \LogicException(sprintf('The "symfony/serializer" component is required to use the "%s" attribute. Try running "composer require symfony/serializer".', Serialize::class));
+        }
+
+        $request = $event->getRequest();
+        $controllerResult = $event->getControllerResult();
+        $format = $request->getPreferredFormat(self::DEFAULT_FORMAT);
```

Comment:
This listener should rather use `$request->getRequestFormat()`, getting the format configured for the response to that request, instead of enforcing usage of negotiation. This would make it compatible with all ways to determine the response format.

Reply:
done